rotkehlchen/chain/ethereum/modules/yearn/graph.py

```diff
@@ -0,0 +1,424 @@
+            from_asset_usd_price = prices_cache.get(from_asset)
+            to_asset_usd_price = prices_cache.get(to_asset)
+
+            if from_asset_usd_price is None:
+                prices_cache[from_asset] = get_usd_price_zero_if_error(
+                    asset=from_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault withdrawal',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                from_asset_usd_price = prices_cache.get(from_asset)
+
+            if to_asset_usd_price is None:
+                prices_cache[to_asset] = get_usd_price_zero_if_error(
+                    asset=to_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault withdrawal',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                to_asset_usd_price = prices_cache.get(to_asset)
+
+            assert to_asset_usd_price is not None
+            assert from_asset_usd_price is not None
+
+            from_asset_amount = token_normalized_value(
+                token_amount=int(entry['sharesBurnt']),
+                token=from_asset,
+            )
+
+            to_asset_amount = token_normalized_value(
+                token_amount=int(entry['tokenAmount']),
+                token=to_asset,
+            )
+
+            result.append(YearnVaultEvent(
+                event_type='withdraw',
+                block_number=int(entry['blockNumber']),
+                timestamp=Timestamp(int(entry['timestamp']) // 1000),
+                from_asset=from_asset,
+                from_value=Balance(
+                    amount=from_asset_amount,
+                    usd_value=from_asset_amount * from_asset_usd_price,
+                ),
+                to_asset=to_asset,
+                to_value=Balance(
+                    amount=to_asset_amount,
+                    usd_value=to_asset_amount * to_asset_usd_price,
+                ),
+                realized_pnl=None,
+                tx_hash=tx_hash,
+                log_index=int(log_index),
+            ))
+
+        return result
+
+    def get_deposit_events(
+        self,
+        addresses: EthAddress,
+        from_block: int,
+        to_block: int,
+    ) -> List[YearnVaultEvent]:
+        param_types = {
+            '$from_block': 'BigInt!',
+            '$to_block': 'BigInt!',
+            '$addresses': '[Bytes!]',
+        }
+
+        param_values = {
+            'from_block': from_block,
+            'to_block': to_block,
+            'addresses': addresses,
+        }
+        query = self.graph.query(
+            querystr=QUERY_USER_DEPOSITS,
+            param_types=param_types,
+            param_values=param_values,
+        )
+        return self._process_deposits(query["deposits"])
+
+    def get_withdraw_events(
+        self,
+        addresses: EthAddress,
```

Comment:
For type of address here and everywhere else please use `ChecksumEthAddress` not EthAddres. Unless ... it's really NOT checksummed.

Reply:
Yes, this address comes from the graph where addresses are not checksummed. When I need to use them in rotki I take care to checksum them